pkg/image/api/helper.go

```diff
@@ -54,20 +54,25 @@ func IsPullSpec(spec string) bool {
 // JoinDockerPullSpec turns a set of components of a Docker pull specification into a single
 // string. Attempts to match as closely as possible the Docker spec up to 1.3. Future API
 // revisions may change the pull syntax.
-func JoinDockerPullSpec(registry, namespace, name, tag string) string {
-	if len(tag) != 0 {
-		tag = ":" + tag
+func JoinDockerPullSpec(registry, namespace, name, ref string) string {
+	if len(ref) != 0 {
```

Comment:
I'm starting to think we need to refactor this into a struct - `PullSpec` with a set of fields, and digest and tag should not be the same.